### PR TITLE
[CARBONDATA-1163][Bug-Fix] Fix CI failures

### PR DESCRIPTION
**Pull request by watermen**
1. Add a flag named needConvertDecimalToByte to determine whether need to convert decimal to byte or not.
2. Fix test case(TestGlobalSortDataLoad.scala) issue on spark-1.6 

@jackylk

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2497/

File: processing/src/main/java/org/apache/carbondata/processing/newflow/sort/unsafe/merger/UnsafeSingleThreadFinalSortFilesMerger.java [hunk 1]
```diff
@@ -259,6 +259,50 @@ public boolean hasNext() {
     return this.fileCounter > 0;
   }
 
+  private Object[] convertRow(Object[] data) {
```
Comment (jackylk) [inline processing/src/main/java/org/apache/carbondata/processing/newflow/sort/unsafe/merger/UnsafeSingleThreadFinalSortFilesMerger.java]: it is better to reuse the same function.
@ravipesala check also

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2500/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2501/

**Comment by jackylk** [comment]
LGTM

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/dataload/TestGlobalSortDataLoad.scala [hunk 2]
```diff
@@ -280,12 +280,13 @@ class TestGlobalSortDataLoad extends QueryTest with BeforeAndAfterEach with Befo
   test("Test with different date types") {
     val path = s"$projectPath/examples/spark2/src/main/resources/data.csv"
 
+    sql("DROP TABLE IF EXISTS carbon_localsort_difftypes")
     sql(
       s"""
          | CREATE TABLE carbon_localsort_difftypes(
-         | shortField SHORT,
+         | shortField smallint,
```
Comment (chenerlu) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/dataload/TestGlobalSortDataLoad.scala]: Suggest SMALLINT for SHORT and BIGINT for LONG

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/dataload/TestGlobalSortDataLoad.scala [hunk 3]
```diff
@@ -280,12 +280,13 @@ class TestGlobalSortDataLoad extends QueryTest with BeforeAndAfterEach with Befo
...
          | intField INT,
-         | bigintField LONG,
+         | bigintField bigint,
```
Comment (chenerlu) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/dataload/TestGlobalSortDataLoad.scala]: bigint -> BIGINT

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/dataload/TestGlobalSortDataLoad.scala [hunk 4]
```diff
@@ -299,16 +300,16 @@ class TestGlobalSortDataLoad extends QueryTest with BeforeAndAfterEach with Befo
...
 
+    sql("DROP TABLE IF EXISTS carbon_globalsort_difftypes")
     sql(
       s"""
          | CREATE TABLE carbon_globalsort_difftypes(
-         | shortField SHORT,
+         | shortField smallint,
          | intField INT,
-         | bigintField LONG,
+         | bigintField bigint,
```
Comment (chenerlu) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/dataload/TestGlobalSortDataLoad.scala]: same problem.

**Comment by ravipesala** [comment]
LGTM

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/378/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-core' /><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/378/org.apache.carbondata$carbondata-core/testReport'>carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-core</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/378/org.apache.carbondata$carbondata-core/testReport/org.apache.carbondata.core.dictionary.client/DictionaryClientTest/testClient/'><strong>org.apache.carbondata.core.dictionary.client.DictionaryClientTest.testClient</strong></a></li></ul>

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/382/<h2>Failed Tests: <span class='status-failure'>3</span></h2><h3><a name='carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-spark-common-test' /><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/382/org.apache.carbondata$carbondata-spark-common-test/testReport'>carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-spark-common-test</a>: <span class='status-failure'>3</span></h3><ul><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/382/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.partition/TestDataLoadingForPartitionTable/data_loading_for_partition_table__range_partition/'><strong>org.apache.carbondata.spark.testsuite.partition.TestDataLoadingForPartitionTable.data loading for partition table: range partition</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/382/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.partition/TestDataLoadingForPartitionTable/Insert_into_for_partition_table__range_partition/'><strong>org.apache.carbondata.spark.testsuite.partition.TestDataLoadingForPartitionTable.Insert into for partition table: range partition</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/382/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.partition/TestQueryForPartitionTable/detail_query_on_partition_table__range_partition/'><strong>org.apache.carbondata.spark.testsuite.partition.TestQueryForPartitionTable.detail query on partition table: range partition</strong></a></li></ul>

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/379/<h2>Failed Tests: <span class='status-failure'>0</span></h2>

**Pull request closed without merging**